### ISSUE TITLE
Installing Window Decoration raise The shared library was not found on Fedora 31

### ISSUE THREAD

**Issue by thobianchi** [n4n0GH/hello#16]
## Installation path error on Fedora 31

```
org.kde.kcoreaddons: Error loading plugin "org.kde.kdecoration2/hellodecoration" "The shared library was not found." 
                                             Plugin search paths are ("/usr/lib64/qt5/plugins", "/usr/bin") 
                                             The environment variable QT_PLUGIN_PATH might be not correctly set
```
I solved this problem moving the shared library from `/usr/lib64/plugins/org.kde.kdecoration2/hellodecoration.so` to `/usr/lib64/qt5/plugins/org.kde.kdecoration2`

**Comment by n4n0GH** [comment]
That's odd.

Can you do me a favor and grab the KDE Breeze theme from here: https://github.com/KDE/breeze/tree/master/kdecoration and build it the same way you would the hello decorations? I'm curious if Breeze shows you the same build error.

**Comment by thobianchi** [comment]
I was not very precise with this reporting. The build process does not give errors:
```
~/Esperimenti/hello-master/window-decoration/build  sudo make install
[  4%] Automatic MOC for target hellocommon5
[  4%] Built target hellocommon5_autogen
[ 16%] Built target hellocommon5
[ 20%] Automatic MOC for target hellodecoration
[ 20%] Built target hellodecoration_autogen
[100%] Built target hellodecoration
Install the project...
-- Install configuration: ""
-- Up-to-date: /usr/lib64/cmake/hello/HelloConfig.cmake
-- Up-to-date: /usr/lib64/cmake/hello/HelloConfigVersion.cmake
-- Up-to-date: /usr/lib64/plugins/org.kde.kdecoration2/hellodecoration.so
-- Up-to-date: /usr/share/kservices5/hellodecorationconfig.desktop
-- Up-to-date: /usr/lib64/libhellocommon5.so.5.16.5
-- Up-to-date: /usr/lib64/libhellocommon5.so.5
```
Since after this I was not seeing hello in widow decoration menu I opened the journal and find that error.

**Comment by n4n0GH** [comment]
Doing a quick websearch, it appears that Fedora indeed uses different paths for this. But the CMake uses ```${PLUGIN_INSTALL_DIR}``` to determine the correct path on the system which leads me to believe that something is bonked on your end.

Can you do me a favor and grab the KDE Breeze theme from here: https://github.com/KDE/breeze/tree/master/kdecoration and build it the same way you would the hello decorations? If my assumption is correct, you'll have the same errors in your journal with this theme.

**Comment by n4n0GH** [comment]
It may also be an issue with an ongoing thing https://bugzilla.redhat.com/show_bug.cgi?id=1233298

TL;DR due to KDE4 compatibility reasons some upstream decisions were made to use certain variables differently which is patched by individual distros, some do it better than others, but apparently this is an ongoing battle.

IF that is truly the cause there's nothing I can do about it, but I'm curious since some other people managed to build this just fine on Fedora.

**Comment by n4n0GH** [comment]
Closing this issue to to inactivity, feel free to continue the discussion and we can re-open this.
